config/base/derive/src/configurable.rs

```diff
@@ -0,0 +1,286 @@
+            .is_ident("config")
+            .then(|| attr.parse_args::<Inner>())
+            .map_or_else(
+                || {
+                    Err(syn::Error::new_spanned(
+                        attr,
+                        "Attribute must be in form #[config...]",
+                    ))
+                },
+                |inner| inner,
+            )
+    }
+}
+
+pub struct ConfigInner {
+    _kw: kw::inner,
+}
+
+impl Parse for ConfigInner {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        Ok(Self {
+            _kw: input.parse()?,
+        })
+    }
+}
+
+pub struct ConfigAsStr {
+    _kw: kw::serde_as_str,
+}
+
+impl Parse for ConfigAsStr {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        Ok(Self {
+            _kw: input.parse()?,
+        })
+    }
+}
+
+pub struct ConfigEnvPrefix {
+    _kw: kw::env_prefix,
+    _eq: Token![=],
+    pub prefix: LitStr,
+}
+
+impl Parse for ConfigEnvPrefix {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        Ok(Self {
+            _kw: input.parse()?,
+            _eq: input.parse()?,
+            prefix: input.parse()?,
+        })
+    }
+}
+
+pub(super) fn impl_configurable(ast: &utils::StructWithFields) -> TokenStream {
+    let utils::StructWithFields {
+        ident: parent_name,
+        vis,
+        generics,
+        ..
+    } = ast;
+    let proxy_name = format_ident!("{}Proxy", parent_name);
+
+    let prefix = utils::get_env_prefix(ast);
+
+    let field_idents = utils::extract_field_idents(&ast.fields);
+
+    let field_attrs = utils::extract_field_attrs(&ast.fields);
+
+    let field_ty = utils::extract_field_types(&ast.fields);
+    let inner = utils::field_has_inner_attr(&field_attrs);
+
+    let field_env = utils::gen_field_env(&field_idents, &prefix);
+
+    let as_str = utils::field_has_as_str_attr(&field_attrs);
+    let option = utils::field_has_option_type(&field_ty);
+
+    let (_lvalue_read, lvalue_write) = utils::gen_lvalues(&field_ty, &field_idents);
+
+    // Doing for parent struct right now instead of the proxy
+    // TODO: when generating for proxy, replace types and option with proxy's values
+    let load_env_impl = impl_load_env(
+        &field_idents,
+        &inner,
+        &option,
+        &lvalue_write,
+        &as_str,
+        &field_ty,
+        &field_env,
+    );
+
+    let disk_impl = impl_load_from_disk();
+    let proxy_fields = gen_proxy_struct_fields(&field_idents, &field_attrs, &field_ty);
+    let build_impl = impl_build(&field_idents, &field_attrs);
+
+    // TODO: decide if attrs are needed here too
+    quote! {
+        /// Proxy configuration structure to be used as an intermediate
+        /// for config loading
+        #[derive(Debug, Default, Clone, serde::Serialize, serde::Deserialize)]
+        // #(#attrs)*
+        #vis struct #proxy_name #generics {
+            #proxy_fields
+        }
+
+        impl iroha_config_base::proxy::LoadFromEnv for #parent_name {
```

Comment:
I would suggest to implemnt `impl_load_from_dist` (and other) in such a way that it will produce whole trait impl.